app/grandchallenge/reader_studies/serializers.py

```diff
@@ -143,6 +143,10 @@ class AnswerSerializer(HyperlinkedModelSerializer):
     answer_image = HyperlinkedRelatedField(
         read_only=True, view_name="api:image-detail"
     )
+    display_set = HyperlinkedRelatedField(
+        queryset=DisplaySet.objects.all(),
```

Comment:
We need to check that the user has permission to create answers for this `display_set` if this field is not read only. This `queryset` (and the one for the images and questions) need to be filtered.

Reply:
Definitely needs to be read_only! I'll add some tests to also cover the other members.

Although I see that those tests are rendering entire views. So those are ready to be optimized first.

Reply:
This is quite tricky, so maybe @MikeOverkamp-diag can help.